tests/commands/test_get.py

```diff
@@ -0,0 +1,465 @@
+import re
+import subprocess
+from pathlib import Path
+from typing import Any, Generator, Union
+
+from onyo.commands.get import (
+    natural_sort, fill_unset, set_filters, sanitize_keys)
+from onyo.lib import Repo, Filter
+import pytest
+
+
+def convert_contents(
+        raw_assets: list[tuple[str, dict[str, Any]]]) -> Generator:
+    """Convert content dictionary to a plain-text string"""
+    for file, raw_contents in raw_assets:
+        contents = ''
+        for k, v in raw_contents.items():
+            if isinstance(v, str):
+                v = f"'{v}'"
+            elif isinstance(v, bool):
+                v = str(v).lower()
+            contents += f'{k}: {v}\n'
+        yield [file, contents]
+
+
+@pytest.mark.repo_contents(*convert_contents([
+    ('laptop_apple_macbookpro.1',
```

Comment:
I don't think that would make sense for `onyo get` to test. It should be able to safely assume that names are valid, because it does not change the repo. If I test for this, then `re.findall()` returns nothing, which breaks the key retrieval. I can test every individual asset for a valid name prior to key retrieval, but if there are invalid assets then `fsck` would already fail.

Reply:
You are right, that the `repo.fsck([asset-yaml])` line tests that, good point!

Just checked this in your PR, and you use just the one fsck, and I suspect that there are other asset related tests that should be run by `get`, too. Especially the one checking that asset names are unique, and the one verifying there are no pseudo keys inside the asset! But I see that some others (like the anchor test) seem not to be needed for onyo get 👍  :)